aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -0,0 +1,1595 @@
+    }
+
+    struct AddStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_added: u64,
+    }
+
+    struct ReactivateStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount: u64,
+    }
+
+    struct UnlockStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_unlocked: u64,
+    }
+
+    struct WithdrawStakeEvent has drop, store {
+        pool_address: address,
+        delegator_address: address,
+        amount_withdrawn: u64,
+    }
+
+    public entry fun initialize_delegation_pool(owner: &signer, operator_fee: u64, delegation_pool_creation_seed: vector<u8>) {
+        let owner_address = signer::address_of(owner);
+        assert!(!owner_cap_exists(owner_address), error::already_exists(EOWNER_CAP_ALREADY_EXISTS));
+        assert!(operator_fee <= MAX_FEE, error::invalid_argument(EINVALID_COMMISSION_PERCENTAGE));
+
+        // generate a seed to be used to create the resource account hosting the delegation pool
+        let seed = vector::empty<u8>();
+        // include module salt (before any subseeds) to avoid conflicts with other modules creating resource accounts
+        vector::append(&mut seed, MODULE_SALT);
+        // include an additional salt in case the same resource account has already been created
+        vector::append(&mut seed, delegation_pool_creation_seed);
+
+        let (stake_pool_signer, stake_pool_signer_cap) = account::create_resource_account(owner, seed);
+        coin::register<AptosCoin>(&stake_pool_signer);
+
+        // stake_pool_signer is owner account of stake pool and has `OwnerCapability`
+        let pool_address = signer::address_of(&stake_pool_signer);
+        stake::initialize_stake_owner(&stake_pool_signer, 0, owner_address, owner_address);
+
+        let inactive_shares = table::new<u64, pool_u64::Pool>();
+        table::add(&mut inactive_shares, 0, pool_u64::create());
+
+        move_to(&stake_pool_signer, DelegationPool {
+            active_shares: pool_u64::create(),
+            observed_lockup_cycle: 0,
+            inactive_shares,
+            pending_withdrawals: table::new<address, u64>(),
+            stake_pool_signer_cap,
+            locked_until_secs: 0,
+            total_coins_inactive: 0,
+            operator_fee,
+            add_stake_events: account::new_event_handle<AddStakeEvent>(&stake_pool_signer),
+            reactivate_stake_events: account::new_event_handle<ReactivateStakeEvent>(&stake_pool_signer),
+            unlock_stake_events: account::new_event_handle<UnlockStakeEvent>(&stake_pool_signer),
+            withdraw_stake_events: account::new_event_handle<WithdrawStakeEvent>(&stake_pool_signer),
+        });
+
+        // save resource-account address (inner pool address) + outer pool ownership on `owner`
+        move_to(owner, DelegationPoolOwnership { pool_address });
```

Comment:
As discussed offline, I think this can be simplified to just storing the owner_address in the delegation pool if there are no immediate use cases for transferrable ownership capability.

Reply:
The decision to be implemented was mainly based on the fact that its function can be disabled for now and easily enabled by adding extract and add capability methods, while having it as a resource on the pool's creator makes it a pointer to the resource account address